core/modules/layout/tests/layout.test

```diff
@@ -1365,6 +1262,244 @@ class LayoutInterfaceTest extends BackdropWebTestCase {
+    $custom_block = $this->xpath('//*[contains(@class,:region)]//*[contains(@class,:block)]', array(
+      ':region' => 'l-sidebar',
+      ':block' => 'block-block-my-custom-block',
+    ));
+    $this->assertEqual(count($custom_block), 1, 'Converted custom block found.');
+    if (count($custom_block)) {
+      $custom_block_title = $custom_block[0]->xpath('.//*[contains(@class, "block-title")]');
+      $custom_block_content = $custom_block[0]->xpath('.//*[contains(@class, "block-content")]/p');
+      $this->assertEqual((string) $custom_block_title[0], $block_title_1, 'Converted custom block title found.');
+      $this->assertEqual((string) $custom_block_content[0], $block_content_1, 'Converted custom block content.');
+    }
+  }
+
+  /**
+   * Test Hero blocks.
+   */
+  function testHeroBlocks() {
+    $hero_title_1 = 'Wonderwoman';
+    $hero_content_1 = 'Youth is your greatest weapon, your greatest tool.';
+    $hero_title_2 = 'Black Widow';
+    $hero_content_2 = "This isnt gonna end well...";
+
+    // Create some sample images.
+    $good_image = imagecreate(2400, 400);
+    $good_filename = 'good.jpg';
+    $good_realpath_filename = backdrop_realpath('temporary://' . $good_filename);
+    imagejpeg($good_image, $good_realpath_filename);
+
+    $bad_image =  imagecreate(1024, 260);
+    imagecolorallocate($bad_image, 255, 0, 0); // Red image.
+    $bad_filename = 'bad.jpg';
+    $bad_realpath_filename = backdrop_realpath('temporary://' . $bad_filename);
+    imagejpeg($bad_image, $bad_realpath_filename);
+
+    $good_image2 = imagecreate(2400, 400);
+    $good_filename2 = 'good2.jpg';
+    $good_realpath_filename2 = backdrop_realpath('temporary://' . $good_filename2);
+    imagejpeg($good_image2, $good_realpath_filename2);
+
+    $this->backdropGet('admin/structure/layouts/manage/default');
+
+    // Block 1: Add a hero block with no image to the content area.
+    $this->clickLink(t('Add block'), 2);
+    $this->clickLink(t('Add a hero block'));
+    $edit = array(
+      'title' => $hero_title_1,
+      'content[value]' => $hero_content_1,
+    );
+    $this->backdropPost(NULL, $edit, t('Add block'));
+
+    // Block 2: Add a hero block with only content and an image.
+    $this->clickLink(t('Add block'), 2);
+    $this->clickLink(t('Add a hero block'));
+    $edit = array(
+      'content[value]' => $hero_content_2,
+      'files[image]' => $good_realpath_filename,
+    );
+    $this->backdropPost(NULL, $edit, t('Add block'));
+
+    // Block 3: Add a hero block with only an image.
+    $this->clickLink(t('Add block'), 2);
+    $this->clickLink(t('Add a hero block'));
+
+    // Confirm a too-small background image fails validation.
+    $edit = array(
+      'files[image]' => $bad_realpath_filename,
+    );
+    $this->backdropPost(NULL, $edit, t('Add block'));
+    $this->assertText(t('The specified file ' . $bad_filename . ' could not be uploaded. The image is too small; the minimum dimensions are 1200x300 pixels.'));
```

Comment:
Copy/paste the actual message from File module and reuse it here rather than concatenating $bad_filename directly into `t()`.